envoy/admin/v2/config_dump.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package envoy.admin.v2;
+
+import "google/protobuf/any.proto";
+import "envoy/api/v2/rds.proto";
+
+import "gogoproto/gogo.proto";
+
+// [#protodoc-title: ConfigDump]
+// [#proto-status: draft]
+
+// The /config_dump admin endpoint uses this wrapper message to maintain and serve arbitrary
+// configuration information from any component in Envoy.
+// TODO(jsedgwick) In the future, we may want to formalize this further with an RPC for config_dump,
+// and perhaps even with an RPC per config type. That strategy across all endpoints will allow for
+// more flexibility w.r.t. protocol, serialization, parameters, etc.
+message ConfigDump {
+  // This map is serialized and dumped in its entirety at the /config_dump endpoint.
+  //
+  // Keys should be a short descriptor of the config object they map to. For example, envoy's HTTP
+  // routing subsystem might use "routes" as the key for its config, for which it uses the message
```

Comment:
I think we should consider making this a type URL in the future. I'm still after a config that could be effectively replayed via an xDS management server.